h/models/user.py

```diff
@@ -70,7 +70,7 @@ class UserIDComparator(Comparator):
             (lower(replace('luis.silva', '.', '')), 'example.com')
     """
 
-    def __init__(self, username, authority):
+    def __init__(self, username, authority):  # pylint: disable=super-init-not-called
         self.username = username
         self.authority = authority
```

Comment:
I'm a bit dubious over this one. Not calling the parent initialiser is only ok here because we happen to implement `__clause_element__` ourselves, but if we called the parent we wouldn't have to.

Also it's presently ok, and I suspect will be for all time, but _theoretically_ SQLAlchemy could change out from underneith us and introduce more behaviors which don't work because we didn't pass up the expression.

I wonder if this would be better as:

```suggestion
    def __init__(self, username, authority):  # pylint: disable=super-init-not-called
        super().__init__(element=sa.tuple_(_normalise_username(username), authority))
```

Then `__clause_element__()` is not required any more, and all references to it can be replaced with `self.element`?

Also this class could really do with something like:

```python
@classmethod
def get_element(cls, username, authority)
    return sa.tuple_(_normalise_username(username, authority)
```

It does it in three separate places.

Reply:
I got a version of this that passes the test but not sure how to manually test the functionality .